pinot-core/src/main/java/org/apache/pinot/core/realtime/impl/RealtimeSegmentConfig.java

```diff
@@ -205,6 +214,18 @@ public PartitionUpsertMetadataManager getPartitionUpsertMetadataManager() {
     return _partitionUpsertMetadataManager;
   }
 
+  public UpsertConfig.STRATEGY getGlobalUpsertStrategy() {
```

Comment:
why do we need three strategies? I thought there is one strategy field, but can be of different types, such as predefined and custom in groovy